osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyMainCirclePiece.cs

```diff
@@ -36,6 +36,42 @@ public LegacyMainCirclePiece(string priorityLookup = null, bool hasNumber = true
         private Container circleSprites;
         private Drawable hitCircleSprite;
         private Drawable hitCircleOverlay;
+        private Drawable aboveNumberProxy;
+        private Container overlayProxy;
+
+        public Drawable OverlayProxy
+        {
+            get
+            {
+                if (overlayProxy == null)
+                {
```

Comment:
Please invert as:
```
if (overlayProxy != null)
    return overlayProxy;
```
In general we try to avoid getters having business logic like this, and using `CreateX()` instead, but I _could_ get around that having this inverted, or having a `createProxy()` method.